test/js/node/fs/fs.windows.test.ts

```diff
@@ -0,0 +1,30 @@
+import fs from "node:fs/promises";
+import { test as testReal, expect } from "bun:test";
+
+const test = process.platform !== "win32" ? testReal.skip : testReal;
+
+{
+  const file = import.meta.path.slice(3);
+  const drive = import.meta.path[0];
+  const filenames = [
+    `${drive}:\\${file}`,
+    `\\\\127.0.0.1\\${drive}$\\${file}`,
+    `\\\\LOCALHOST\\${drive}$\\${file}`,
+    `\\\\.\\${drive}:\\${file}`,
+    `\\\\?\\${drive}:\\${file}`,
+    `\\\\.\\UNC\\LOCALHOST\\${drive}$\\${file}`,
+    `\\\\?\\UNC\\LOCALHOST\\${drive}$\\${file}`,
+    `\\\\127.0.0.1\\${drive}$\\${file}`,
+  ];
+
+  for (const filename of filenames) {
+    test(`Can read '${filename}' with node:fs`, async () => {
```

Comment:
we should just do `test.if(isWindows)(..)` so that we don't have an ever-growing number of ways in which we do platform-specific tests.